src/Elasticsearch.Extensions.Logging/ElasticsearchLoggerOptions.cs

```diff
@@ -68,6 +69,45 @@ public class ElasticsearchLoggerOptions
 		/// 'Production', etc).
 		/// </summary>
 		public string[] Tags { get; set; } = new string[0];
+
+		public DrainThrottles Throttles { get; set; } = new DrainThrottles();
+
+	}
+
+	/// <summary>
+	/// Controls how <see cref="LogEvent"/>'s are batched and send to Elasticsearch. These can not be dynamically updated.
+	/// </summary>
+	public class DrainThrottles
+	{
+		/// <summary>
+		/// The maximum number of <see cref="LogEvent"/> that can be queued in memory. If this threshold is reached, events will be dropped
+		/// </summary>
+		public int MaxInFlightMessages { get; set; } = 100_000;
+
+		/// <summary>
+		/// The number of events a local buffer should reach before sending the events in a single call to Elasticsearch.
+		/// </summary>
+		public int MaxConsumerBufferSize { get; set; } = 1_000;
+
+		/// <summary>
+		/// A consumer builds up a local buffer until <see cref="MaxConsumerBufferSize"/> is reached. If events come in too slow, these
+		/// events could end up taking forever to be sent to Elasticsearch. This controls how long a buffer may exist before a flush is triggered.
+		/// </summary>
+		public TimeSpan MaxConsumerBufferLifetime { get; set; } = TimeSpan.FromSeconds(5);
+
+		/// <summary>
+		/// The maximum number of consumers allowed to poll for new events on the channel. Defaults to 1, increase to introduce concurrency.
+		/// </summary>
+		public int ConcurrentConsumers { get; set; } = 1;
+
+		/// <summary>
+		/// If <see cref="MaxInFlightMessages"/> is reached, <see cref="LogEvent"/>'s will fail to be published to the channel. You can be notified of dropped
+		/// events with this callback
+		/// </summary>
+		public Action<LogEvent> PublishRejectionCallback { get; set; } = e => { };
+
+		public Action<IElasticsearchResponse, IConsumerBuffer> ElasticsearchResponseCallback { get; set; } = (r, b) => { };
```

Comment:
This makes sense since you might want to subscribe multiple times in different layers of your application. At the very least we don't know if folks want to always ONLY have 1 listener.

The callbacks are not used for control flow either. Added a TODO to follow up on in a dedicated PR.